### PR TITLE
Login

### PR DESCRIPTION
**Pull request by ekat-br**
This is the PR for [US11](https://github.com/ekat-br/roomcheck/issues/24). Thanks in advance for reviewing my code. 
Credentials for testing can be found in my code 🤓

### CODE REVIEW

File: components/HomepageNavigation/index.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+        <NavButton>
+          <HomeIcon />
+        </NavButton>
+      </Link>
+      <Link href="/login">
+        <NavButton>
+          <LoginIcon />
+        </NavButton>
+      </Link>
+    </NavContainer>
```
Comment (julialoeschel) [inline components/HomepageNavigation/index.js]: you dont need to but a button inside a Link component, this is very confusing and bad a11y. Next Suggests a way like this: 
```
 
function Home() {
  return (
    <ul>
      <li>
        <Link href="/">Home</Link>
      </li>
      <li>
        <Link href="/about">About Us</Link>
      </li>
      <li>
        <Link href="/blog/hello-world">Blog Post</Link>
      </li>
    </ul>
  )
}
 
export default Home
```

File: components/Login/index.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+export default function Login() {
+  const router = Router;
+  const { data: session } = useSession();
+
+  if (session) {
+    return (
+      <LoginContainer>
+        <LoginText>Eingeloggt als {session.user.name}</LoginText>
+        <LogoutButton onClick={signOut}>Log Out</LogoutButton>
+        <AdminButton onClick={() => router.push("/admin")}>
```
Comment (julialoeschel) [inline components/Login/index.js]: since you are Linking to another page in your app, This should be a Link.

File: pages/admin/index.js [hunk 3]
```diff
@@ -1,24 +1,78 @@
...
+      <>
+        <Heading>🏫✅ RoomCheck ✅🏫</Heading>
+        <AdminViewContainer>
+          <Heading>Zugriff verweigert</Heading>
+          <Link href="/login">
+            <AdminButton>Login</AdminButton>
+          </Link>
+          <Link href="/">
+            <HomeButton>Startseite</HomeButton>
+          </Link>
```
Comment (julialoeschel) [inline pages/admin/index.js]: Same Link-Button- Issue here.
try to semantically restructure it to generate a better codebase.

File: pages/login/index.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import Heading from "../../components/Heading";
+import HomePageNavigation from "../../components/HomepageNavigation";
+import Login from "../../components/Login";
+
+export default function LoginPage() {
+  return (
+    <div>
```
Comment (julialoeschel) [inline pages/login/index.js]: this could be a fragment <></>

**Review by julialoeschel — COMMENTED** [review]
really good job there. 
I have some suggestions, especially in semantics. 
Please take a minute to find some information about Links and buttons, to improve your structure in the future. 

The authentication looks great, really good job!

File: components/HomepageNavigation/index.js (hunk 1 above)
Comment (ekat-br) [reply components/HomepageNavigation/index.js]: Hi Julia, to be honest I am not convinced. This is what I learned in my course. Can you explain further why this is not possible?

File: components/Login/index.js (hunk 2 above)
Comment (ekat-br) [reply components/Login/index.js]: Thx, changed it to this: `<Link href="/admin">
          <AdminButton>Go to Admin View</AdminButton>
        </Link>`

File: pages/admin/index.js (hunk 3 above)
Comment (ekat-br) [reply pages/admin/index.js]: I'm sorry, but I cannot understand why this is not allowed.

File: pages/login/index.js (hunk 4 above)
Comment (ekat-br) [reply pages/login/index.js]: Changed it

File: components/HomepageNavigation/index.js (hunk 1 above)
Comment (Roland-Hufnagel) [reply components/HomepageNavigation/index.js]: Who told you that? 😀😀😀
You are mixing two principles. Either use a button OR use a link. The reason is accessabillity: They are semantically different and screenreaders / searchengines couldn't interpret a button inside a link. 
And I cannot see any reason for your solution. You can style the Link like a button and wrap it around the icon.

File: components/Login/index.js (hunk 2 above)
Comment (Roland-Hufnagel) [reply components/Login/index.js]: Why do you like that 'button in a link' so much? 😀

**Review by Roland-Hufnagel — CHANGES_REQUESTED** [review]
I left some comments. Hope this is helpful.

File: components/HomepageNavigation/index.js (hunk 1 above)
Comment (ekat-br) [reply components/HomepageNavigation/index.js]: I wish I could remember 😂  I think I heard it at some point from a coach. Maybe it was also said that it should NOT be done that way. Ooops. I wish someone had stopped me earlier

Comment (ekat-br) [reply components/HomepageNavigation/index.js]: How do you like this?  
```
import { NavButton, NavContainer, HomeIcon, LoginIcon } from "./styles";
import { useRouter } from "next/router";

export default function HomePageNavigation() {
  const router = useRouter();
  const handleHomeClick = () => {
    router.push("/");
  };
  const handleLoginClick = () => {
    router.push("/login");
  };
  return (
    <NavContainer>
      <NavButton onClick={handleHomeClick}>
        <HomeIcon />
      </NavButton>
      <NavButton onClick={handleLoginClick}>
        <LoginIcon />
      </NavButton>
    </NavContainer>
  );
}
```

File: components/Login/index.js (hunk 2 above)
Comment (ekat-br) [reply components/Login/index.js]: I think I just liked the fact that it worked. But now I finally have to let it go. Changed it.

File: pages/admin/index.js (hunk 3 above)
Comment (ekat-br) [reply pages/admin/index.js]: Changed it :)

**Comment by ekat-br** [comment]
Now I see the serious mistake. Thank you for the feedback. I have made the desired changes.

**Review by Roland-Hufnagel — APPROVED** [review]
It is okay now. But the other way is the way, I prefer. As Julia said:
Since you just "link" to another page in your app it is semantically best to use a "Link" instead of a Button.

**Pull request merged**